TAGradingServer/account/admin-gradeable.php

```diff
@@ -1574,6 +1575,75 @@ function moveQuestionUp(question) {
         currentRow.children()[1].children[1].checked = newRow.children()[child].children[1].checked;
         newRow.children()[child].children[1].checked = temp;
     }
+
+    //checks the form to see if it is valid
+    function checkForm()
+    {
+        var gradeable_id = $('#gradeable_id').val();
+        var date_submit = $('#date_submit').val();
+        var date_due = $('#date_due').val();
+        var date_ta_view = $('#date_ta_view').val();
+        var date_grade = $('#date_grade').val();
+        var date_released = $('#date_released').val();
+        var config_path = $('input[name=config_path]').val();
+        var has_space = gradeable_id.includes(" ");
+        var test = /^[a-zA-Z0-9_]*$/.test(gradeable_id);
+        var unique_gradeable = false;
+        var check1 = document.getElementById('radio_electronic_file').checked;
+        var check2 = document.getElementById('radio_checkpoints').checked;
+        var check3 = document.getElementById('radio_numeric').checked;
+
+        if (!test || has_space || gradeable_id == "" || gradeable_id === null) {
+            alert("Gradeable id cannot be blank, null, have a space, contain special characters and must be unique");
+            return false;
+        }
```

Comment:
Change the alert to be "Gradeable ID must not be blank and only contain characters __a-z__ __A-Z__ __0-9__ __\___ __-__" (note emphasis).

@bmcutler do we want to accept periods as well?